components/script/dom/servoparser/mod.rs

```diff
@@ -943,49 +946,70 @@ impl FetchResponseListener for ParserContext {
         self.parser = Some(Trusted::new(&*parser));
         self.submit_resource_timing();
 
-        let content_type = match content_type {
-            Some(ref content_type) => content_type,
-            None => {
-                // No content-type header.
-                // Merge with #4212 when fixed.
-                return;
-            },
+        // Steps for https://html.spec.whatwg.org/multipage/#loading-a-document
+        //
+        // Step 1. Let type be the computed type of navigationParams's response.
+        let mime_type = MimeClassifier::default().classify(
+            LoadContext::Browsing,
+            NoSniffFlag::Off,
+            ApacheBugFlag::from_content_type(content_type.as_ref()),
+            &content_type,
+            // TODO(14024): Figure out how to pass the response data here for sniffing
+            // Requires implementation of https://mimesniff.spec.whatwg.org/#read-the-resource-header
+            &[],
```

Comment:
We're going to need to move a bunch of this logic out of process_response and into process_response_chunk to support this. Fun!

Reply:
Yup. Will do that in a follow-up to keep things manageable.